src/dimension.c

```diff
@@ -379,16 +401,72 @@ dimension_tuple_update(TupleInfo *ti, void *data)
 	return false;
 }
 
+static void
+dimension_insert_relation(Relation rel, int32 hypertable_id,
+						  Name colname, Oid coltype, int16 num_slices,
+						  regproc partitioning_func, int64 interval_length)
+{
+	TupleDesc	desc = RelationGetDescr(rel);
+	Datum		values[Natts_dimension];
+	bool		nulls[Natts_dimension] = {false};
+	CatalogSecurityContext sec_ctx;
+
+	values[Anum_dimension_hypertable_id - 1] = Int32GetDatum(hypertable_id);
+	values[Anum_dimension_column_name - 1] = NameGetDatum(colname);
+	values[Anum_dimension_column_type - 1] = ObjectIdGetDatum(coltype);
+
+	if (OidIsValid(partitioning_func))
+	{
+		Form_pg_proc procform = get_procform(partitioning_func);
+		const char *nspname = get_namespace_name(procform->pronamespace);
+
+		values[Anum_dimension_partitioning_func - 1] = NameGetDatum(&procform->proname);
+		values[Anum_dimension_partitioning_func_schema - 1] = CStringGetDatum(nspname);
```

Comment:
So, yes, I can add the conversion. However, I actually intentionally did it this way since `pronamespace` is read from `pg_catalog` and is thus guaranteed to be within `NAMEDATALEN`. Conversion is thus unnecessary. Besides, if it wouldn't be, the scan would just fail. How do we want to deal with these cases?

Reply:
Or was this about muting Valgrind warnings? Anyway, I'll fix.